src/User/UserUniverseDiscoveryService.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EtoA\User;
+
+use EtoA\Core\Configuration\ConfigurationService;
+
+class UserUniverseDiscoveryService
+{
+    private ConfigurationService $config;
+    private UserRepository $userRepository;
+
+    public function __construct(
+        ConfigurationService $config,
+        UserRepository $userRepository
+    ) {
+        $this->config = $config;
+        $this->userRepository = $userRepository;
+    }
+
+    private function loadDiscoveryMask(int $userId): string
+    {
+        $discoveryMask = $this->userRepository->getDiscoverMask($userId);
+        if (!filled($discoveryMask) || strlen($discoveryMask) < 3) {
+            $sx_num = $this->config->param1Int('num_of_sectors');
+            $cx_num = $this->config->param1Int('num_of_cells');
+            $sy_num = $this->config->param2Int('num_of_sectors');
+            $cy_num = $this->config->param2Int('num_of_cells');
+
+            $mask = str_repeat('0', $sx_num * $cx_num * $sy_num * $cy_num);
+            $this->userRepository->saveDiscoveryMask($userId, $mask);
+
+            return $mask;
+        }
+
+        return $discoveryMask;
+    }
+
+    public function discovered(int $userId, int $absX, int $absY): bool
+    {
+        $sy_num = $this->config->param2Int('num_of_sectors');
+        $cy_num = $this->config->param2Int('num_of_cells');
+
+        $mask = $this->loadDiscoveryMask($userId);
```

Comment:
This now fetches the discovery mask for every field. Wonder whether instead we should pass in a User object or some sort of context

Reply:
done 
6577bec